### PR TITLE
Set custom base directory

### PR DESCRIPTION
**Pull request by assaf13**
Fixed https://onejira.imperva.com/browse/EDSF-424

### CODE REVIEW

File: modules/aws/sonar-base-instance/setup.tftpl [hunk 1]
```diff
@@ -278,4 +284,4 @@ fi
 
 set_environment_vars
 install_ssh_keys
-install_access_tokens
```
Comment (lindanasredin) [inline modules/aws/sonar-base-instance/setup.tftpl:281]: So there is no change here?

File: modules/aws/sonar-base-instance/setup.tftpl [hunk 2]
```diff
@@ -258,11 +258,17 @@ __EOF__
     fi
 }
 
-STATE_DIR=/imperva
-DATA_DIR=$STATE_DIR/data
-LOGS_DIR=$STATE_DIR/logs
-LOCAL_DIR=$STATE_DIR/local
-APPS_DIR=$STATE_DIR/apps
+if [ -z "${base_directory}" ]; then
+  DATA_DIR=/data
```
Comment (lindanasredin) [inline modules/aws/sonar-base-instance/setup.tftpl]: No need for special handling when base_directory is empty, the code in the 'else' is enough

File: examples/installation/dsf_single_account_deployment/variables.tf [hunk 3]
```diff
@@ -448,6 +448,12 @@ variable "sonarw_gw_public_key_file_path" {
   description = "The Agentless Gateway sonarw user public key file path - used for remote Agentless Gateway federation, HADR, etc."
 }
 
+variable "sonar_machine_base_directory" {
```
Comment (lindanasredin) [inline examples/installation/dsf_single_account_deployment/variables.tf:451]: Nice, I agree that this variable should be added to all installation examples but not to poc

Comment (assaf13) [reply examples/installation/dsf_single_account_deployment/variables.tf:451]: Thank you

File: modules/aws/sonar-base-instance/setup.tftpl (hunk 1 above)
Comment (assaf13) [reply modules/aws/sonar-base-instance/setup.tftpl:281]: I removed the redudant new line

File: modules/aws/sonar-base-instance/setup.tftpl (hunk 2 above)
Comment (assaf13) [reply modules/aws/sonar-base-instance/setup.tftpl]: Done

**Review by lindanasredin — APPROVED** [review]

**Pull request merged**